Lib/axisregistry/__init__.py

```diff
@@ -1,52 +1,458 @@
 from google.protobuf import text_format
-
+from collections import OrderedDict
 from axisregistry.axes_pb2 import AxisProto
+from collections import defaultdict
+from itertools import chain
+import logging
+from glob import glob
+import os
 
 try:
     from ._version import version as __version__  # type: ignore
 except ImportError:
     __version__ = "0.0.0+unknown"
 
+log = logging.getLogger(__file__)
+
+# TODO we may have more of these. Please note that some applications may not
+# implement variable font style linking.
+LINKED_VALUES = {
+    "wght": {400.0: 700.0},
+    "ital": {0.0: 1.0},
+}
+
+# Static font styles. The GF api only support the following static font styles
+GF_STATIC_STYLES = OrderedDict(
+    [
+        ("Thin", 100),
+        ("ExtraLight", 200),
+        ("Light", 300),
+        ("Regular", 400),
+        ("Medium", 500),
+        ("SemiBold", 600),
+        ("Bold", 700),
+        ("ExtraBold", 800),
+        ("Black", 900),
+        ("Thin Italic", 100),
+        ("ExtraLight Italic", 200),
+        ("Light Italic", 300),
+        ("Italic", 400),
+        ("Medium Italic", 500),
+        ("SemiBold Italic", 600),
+        ("Bold Italic", 700),
+        ("ExtraBold Italic", 800),
+        ("Black Italic", 900),
+    ]
+)
+
+
+def load_protobuf(klass, path):
+    message = klass()
+    with open(path, "rb") as text_data:
+        text_format.Merge(text_data.read(), message)
+    return message
+
+
+class AxisRegistry:
+    def __init__(self, fp=resource_filename("axisregistry", "data")):
+        axis_fps = [fp for fp in glob(os.path.join(fp, "*.textproto"))]
+        self._data = {}
+        for fp in axis_fps:
+            axis = load_protobuf(AxisProto, fp)
+            self._data[axis.tag] = axis
+
+    def __getitem__(self, k):
+        return self._data[k]
+
+    def __iter__(self):
+        for i in self._data:
+            yield i
+
+    def get_fallback(self, name):
+        for a in self:
+            for fallback in self[a].fallback:
+                if name == fallback.name:
+                    return a, fallback
+        return None, None
+
+    def fallbacks_in_fvar(self, ttFont):
+        res = defaultdict(list)
+        axes_in_font = {
+            a.axisTag: {"min": a.minValue, "max": a.maxValue}
+            for a in ttFont["fvar"].axes
+        }
+        for axis in self:
+            if axis not in axes_in_font:
+                log.warn(f"Axis {axis} not found in GF Axis Registry!")
+                continue
+            for fallback in self[axis].fallback:
+                if (
+                    fallback.value < axes_in_font[axis]["min"]
+                    or fallback.value > axes_in_font[axis]["max"]
+                ):
+                    continue
+                res[axis].append(fallback)
+        return res
+
+    def fallbacks_in_name_table(self, ttFont):
+        res = []
+        name_table = ttFont["name"]
+        tokens = (
+            name_table.getBestFamilyName().split()
```

Comment:
Are you sure you want to pass the *family name* through the fallback process? What happens to a font called "Wonky"?

Reply:
Family names may contain style tokens e.g Cabin Condensed. I'll make a test using your example to see what the repercussions are.

Reply:
I've taken a stab at this. We now skip the first family name token and when we're building the vf stylename, we make sure that it only contains tokens which are not in the family name. Not perfect by any means but covers most cases. If you're still not happy with this, I'll go back to the drawing board.